main.go

```diff
@@ -72,3 +72,55 @@ func stripPath(path string) string {
 	dir := filepath.Dir(path)
 	return strings.TrimPrefix(dir, build.Default.GOPATH+"/src/")
 }
+
+// safeRename safely moves files across different file systems.
+func safeRename(src, dest string) error {
+	if err := os.Rename(src, dest); err != nil {
+		if _, ok := err.(*os.LinkError); ok {
+			// Looks like we are trying to move files across system volumes. Try to copy/delete strategy.
+
+			// Create a temp file which name is guaranteed not to colide with the existing files.
+			tmpFile, err := ioutil.TempFile(filepath.Dir(dest), filepath.Base(dest))
+			if err != nil {
+				return err
+			}
+			tmpPath := tmpFile.Name() // capture path
+			tmpFile.Close()           // do not keep an open file handle
+			defer os.Remove(tmpPath)  // Always remove temporary file
+
+			// Get source file permissions
```

Comment:
Shouldn't you do read operations before write operations? This should be before the TempFile() call above.